src/main/java/com/datatorrent/netlet/WriteOnlyClient.java

```diff
@@ -0,0 +1,239 @@
+    try {
+      lock.lock();
+      isWriteEnabled = !isWriteSuspended();
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public void unregistered(SelectionKey key)
+  {
+    try {
+      lock.lock();
+      if (sendQueue.peek() == null && !isWriteEnabled) {
+        super.unregistered(key);
+      } else {
+        sendQueue = new ForwardingQueue(sendQueue)
+        {
+          @Override
+          public boolean offer(Slice f)
+          {
+            throw new RuntimeException("Client " + this + " does not accept new data.");
+          }
+        };
+      }
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public final void read() throws IOException
+  {
+    if (suspendReadIfResumed()) {
+      logger.warn("{} OP_READ should be disabled", this);
+    } else {
+      logger.error("{} read is not expected", this);
+    }
+  }
+
+  @Override
+  public void write() throws IOException
+  {
+    /*
+     * at first when we enter this function, our buffer is in fill mode.
+     */
+    int remaining = writeBuffer.remaining();
+    Slice slice;
+    while ((slice = sendQueue.peek()) != null) {
+      while (remaining < slice.length) {
+        if (remaining > 0) {
+          writeBuffer.put(slice.buffer, slice.offset, remaining);
+          slice.offset += remaining;
+          slice.length -= remaining;
+        }
+        if (channelWrite() == 0) {
+          return;
+        }
+        remaining = writeBuffer.remaining();
+      }
+      writeBuffer.put(slice.buffer, slice.offset, slice.length);
+      slice.buffer = null;
+      remaining -= slice.length;
+      freeQueue.offer(sendQueue.poll());
+    }
+    channelWrite();
+  }
+
+  protected int channelWrite() throws IOException
+  {
+    if (writeBuffer.flip().remaining() > 0) {
+      final SocketChannel channel = (SocketChannel)key.channel();
+      final int write = channel.write(writeBuffer);
+      writeBuffer.compact();
+      return write;
+    } else {
+      writeBuffer.clear();
+      try {
+        lock.lock();
+        if (sendQueue.peek() == null) {
+          suspendWriteIfResumed();
+          isWriteEnabled = false;
+        }
+      } finally {
+        lock.unlock();
+      }
+      return 0;
+    }
+  }
+
+  public boolean send(byte[] array)
+  {
+    return send(array, 0, array.length);
+  }
+
+  public boolean send(byte[] array, int offset, int len)
+  {
+    /*
+    if (!throwables.isEmpty()) {
+      NetletThrowable.Util.throwRuntime(throwables.pollUnsafe());
+    }
+    */
+    Slice f = freeQueue.poll();
+    if (f == null) {
+      f = new Slice(array, offset, len);
+    } else {
+      if (f.buffer != null) {
+        throw new RuntimeException("Unexpected slice " + f.toString());
+      }
+      f.buffer = array;
+      f.offset = offset;
+      f.length = len;
+    }
+    return send(f);
+  }
+
+  public boolean send(Slice f)
+  {
+    if (sendQueue.offer(f)) {
+      try {
+        lock.lock();
```

Comment:
checking !isWriteEnabled will require declaring isWriteEnabled as volatile, but it would be necessary to use the lock for atomicity anyway.